### PR TITLE
feat: add to_numpy() for Date and support open intervals

### PR DESCRIPTION
**Pull request by clausmichele**
Added method `to_numpy()` for class `Date`, since the base class `datetime.datetime` is the same as for `DateTime`, where `to_numpy()` was already implemented.

Added the possibility to have a `None` value in a `TemporalInterval`, which allows for open ended intervals.

### CODE REVIEW

File: tests/test_pg_parser.py
Comment (LukeWeidenwalker) [inline tests/test_pg_parser.py:248]:
```suggestion
    assert isinstance(parsed_arg.__root__, datetime.datetime)
    assert parsed_arg.to_numpy() == np.datetime64(argument_valid["date"])
```

To test this too!

File: openeo_pg_parser_networkx/pg_schema.py
Comment (LukeWeidenwalker) [inline openeo_pg_parser_networkx/pg_schema.py:244]: We should add a test to make sure this behaves correctly to test_temporal_intervals too!

**Review by LukeWeidenwalker — CHANGES_REQUESTED** [review]
Thanks, this looks almost ready, two small comments!

File: openeo_pg_parser_networkx/pg_schema.py
Comment (clausmichele) [reply openeo_pg_parser_networkx/pg_schema.py:244]: Ok I added a test for an open ended temporal interval!

**Comment by LukeWeidenwalker** [comment]
Thanks for adding the tests - this is looking good now! Seems like it's only the precommit hooks that are failing here - usually I'd fix that on the spot, but I don't have write access on your fork - happy to merge once that runs through! :)

**Comment by LukeWeidenwalker** [comment]
I was just surprised to learn that I can push to your fork, so just fixed the precommit! :D 
Thanks for this!

**Review by LukeWeidenwalker — APPROVED** [review]

**Pull request merged**
